storage/src/factory.rs

```diff
@@ -112,8 +112,9 @@ impl BlobFactory {
         let key = BlobCacheMgrKey {
             config: config.clone(),
         };
+        let mut guard = self.mgrs.lock().unwrap();
```

Comment:
Looks not necessary to hold the mutex while creating each cache object where blob.meta is being downloaded.
I prefer the original style

Reply:
for the original style, if we call new_blob_cache() with same config simultaneously , we will create multiple BlobCacheMgr